index.js

```diff
@@ -522,6 +522,27 @@
         });
       });
 
+  //# foldl :: (a -> b -> a) -> a -> [b] -> a
+  //.
+  //. Returns a single value by iterating through a list, successively
+  //. calling an iterator function and passing it an accumulator value and
+  //. the current value from the list, and then passing the result to the
+  //. next call.
+  //.
+  //. ```javascript
+  //. > S.foldl(S.add, 0, [1, 2, 3, 4, 5])
+  //. 15
+  S.foldl =
+  def('foldl',
+      {},
+      [$.Function, a, List(b), a],
```

Comment:
I believe we want `$.Array(b)` rather than `List(b)`: `List` exists to support functions such as [`S.head`][1] which operate on arrays or strings. We should use `$.Array` rather than `List` in the definitions of `S.pipe` and `S.meld` as well.


[1]: https://github.com/plaid/sanctuary#head